iothub/device/src/ModernDotNet/HsmAuthentication/Transport/HttpUdsMessageHandler.cs

```diff
@@ -42,9 +42,14 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
 
         private async Task<Socket> GetConnectedSocketAsync()
         {
-            var endpoint = new UnixDomainSocketEndPoint(_providerUri.LocalPath);
             Socket socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+#if NET451 || NET472 || NETSTANDARD2_0
+            var endpoint = new UnixDomainSocketEndPoint(_providerUri.LocalPath);
             await socket.ConnectAsync(endpoint).ConfigureAwait(false);
+#else
+            var endpoint = new System.Net.Sockets.UnixDomainSocketEndPoint(_providerUri.LocalPath);
+            await socket.ConnectAsync(endpoint).ConfigureAwait(false);
```

Comment:
L48 and L51 call the same API with the same params, it can be moved out of the if-else block.

Reply:
I take that back, it is an instance of these types that have different implementations for different frameworks.